source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -104,8 +104,29 @@ void CacheFilter::onHeaders(LookupResult&& result) {
       state_ = GetHeadersState::GetHeadersResultUnusable;
     }
     return;
-  case CacheEntryStatus::SatisfiableRange: // TODO(#10132): break response content to the ranges
-                                           // requested.
+  case CacheEntryStatus::NotSatisfiableRange:
+    result.headers_->setStatus(static_cast<uint64_t>(Http::Code::RangeNotSatisfiable));
+    result.headers_->setContentLength(0);
+    result.headers_->addCopy(Http::Headers::get().ContentRange,
+                             absl::StrCat("bytes */", result.content_length_));
+    result.content_length_ = 0;
+  case CacheEntryStatus::SatisfiableRange:
+    // Multi-part responses are not supported; they will be treated as usual 200 response on '::Ok'
+    // case below. A possible way to achieve that would be to move all ranges to remaining_body, and
+    // add logic inside '::onBody' to interleave the body bytes with sub-headers and separator
+    // string for each part. Would need to keep track if the current range is over or not to know
+    // when to insert the separator, and calculate the length based on length of ranges + extra
+    // headers and separators.
+    if (result.response_ranges_.size() == 1) {
+      remaining_body_ = std::move(result.response_ranges_);
+      result.headers_->setStatus(static_cast<uint64_t>(Http::Code::PartialContent));
+      result.headers_->setContentLength(remaining_body_[0].length());
+      result.headers_->addCopy(Http::Headers::get().ContentRange,
+                               absl::StrCat("bytes ", remaining_body_[0].begin(), "-",
+                                            remaining_body_[0].end() - 1, "/",
+                                            result.content_length_));
+      result.content_length_ = remaining_body_[0].length();
```

Comment:
It's confusing to modify result.content_length_ to communicate to other bits of code. Using a helper function instead of fallthrough should make it easy to avoid needing such subtleties. In general, it would be clearer to avoid modifying result at all, except by moving from it. That way it's clear that result represents what we found in the cache (until we make it meaningless by moving from it). It's usually best to avoid making a variable mean one thing and then another.

Reply:
I was already working on the helper functions from the other comments above.

I hadn't consider what you said about moving the data out of result to have these clear separate meanings though, and it makes a lot of sense!
I'll work that into my changes, thanks.